lib/tasks/export.rake

```diff
@@ -0,0 +1,200 @@
+# -*- coding: utf-8 -*-
+#rake tasks and supporting models and functions to do research export
+namespace :export do
+
+require 'csv'
+require 'fileutils'
+
+#create models to access join and translation tables
+class InfoRequestBatchPublicBody < ActiveRecord::Base
+  self.table_name = "info_request_batches_public_bodies"
+  belongs_to :info_request_batch
+  belongs_to :public_body
+end
+
+class PublicBodyCategoryTranslation < ActiveRecord::Base
+  self.table_name = "public_body_category_translations"
+  belongs_to :public_body_category
+end
+
+class PublicBodyHeadingTranslation < ActiveRecord::Base
+  self.table_name = "public_body_category_translations"
+  belongs_to :public_body_heading
+end
+
+class HasTagStringTag < ActiveRecord::Base
+  self.table_name = "has_tag_string_tags"
+end
+
+
+# Remove all instances of user's name (if there is a user), otherwise
+#  return the original text unchanged
+#
+# text - the raw text that needs redaction
+# user - the user object (may be nil)
+#
+# Returns a String
+def case_insensitive_user_censor(text, user)
+  if user && text
+    text.gsub(/#{user.name}/i, "<REQUESTER>")
+  else
+    text
+  end
+end
+
+# Returns a lambda to pass to export function that censors x.property
+def name_censor_lambda(property)
+  lambda do |x|
+    case_insensitive_user_censor(x.send(property), x.info_request.user)
+  end
+end
+
+
+# Exports a model
+#
+# query    - a query used to limit the export to matching records
+# header   - used to restrict exported columns
+# override - pass in lambdas to modify a given column based on values in the row
+#
+# Returns a String
+def csv_export(model, query=nil, header=nil, override={})
+  # set query and header to default values unless supplied
+  query = model.all unless query
```

Comment:
oh, wait - this breaks for `InfoRequestBatchPublicBody` as it doesn't have a primary key